nucypher/blockchain/eth/agents.py

```diff
@@ -574,6 +581,38 @@ class CoordinatorAgent(EthereumContractAgent):
     contract_name: str = "Coordinator"
     _proxy_name = None
 
+    class G2Point(NamedTuple):
```

Comment:
Why is this class necessary?

Reply:
Getting/setting the provider public key.